lib/ansible/plugins/connection/__init__.py

```diff
@@ -360,7 +360,8 @@ def set_implementation_plugin_options(self, task_keys=None, var_options=None, di
         initialize implementation plugin options
         '''
         for plugin in self._implementation_plugins:
-            plugin.set_options(task_keys=task_keys, var_options=var_options, direct=direct)
+            if hasattr(plugin, 'set_options'):
+                plugin.set_options(task_keys=task_keys, var_options=var_options, direct=direct)
```

Comment:
just cause it is 'slightly' more efficient for the 'happy path', put in try/catch and just use set_options, then capture and ignore the attribute error